corehq/motech/fhir/tasks.py

```diff
@@ -135,6 +144,31 @@ def claim_service_request(requests, service_request, case_id):
         response.raise_for_status()
 
 
+def build_case_block(resource_type, resource, suggested_case_id):
+    domain = resource_type.domain
+    case_type = resource_type.case_type.name
+    owner_id = resource_type.fhir_importer.owner_id
+    case = None
+
+    case_id = get_case_id_or_none(resource)
+    external_id = resource['id'] if 'id' in resource else CaseBlock.undefined
+    if case_id:
+        case = get_case_by_id(domain, case_id)
+    if case is None and external_id != CaseBlock.undefined:
+        case = get_case_by_external_id(domain, external_id, case_type)
```

Comment:
These behaviors seem suspect to me - `get_case_by_id` and `get_case_by_external_id` can both return `None` if the case isn't found.  This could be something like an external ID being changed, or someone mixing up two domains.  I think we'd want to fail hard in those scenarios rather than quietly create a new case.  I'm not positive about that, but want to be sure the resolution of those scenarios is deliberate.